src/compiler/program.ts

```diff
@@ -2883,13 +2883,32 @@ namespace ts {
             }
         }
 
+        function pathForLibFile(libFileName: string): string {
+            // Support resolving to lib.dom.d.ts -> @typescript/dom, and
+            //                      lib.dom.iterable.d.ts -> @typescript/dom/iterable
+            //                      lib.es2015.symbol.wellknown.d.ts -> @typescript/es2015/symbol-wellknown
+            const components = libFileName.split(".");
+            let path = components[1];
+            let i = 2;
+            while (components[i] && components[i] !== "d") {
+                path += (i === 2 ? "/" : "-") + components[i];
+                i++;
+            }
+            // The strange "/fake.file" is to ensure we first resolve in node_modules for the current directory.
+            const localOverride = resolveModuleName("@typescript/" + path, currentDirectory + "/fake.file", { moduleResolution: ModuleResolutionKind.NodeJs }, host, moduleResolutionCache);

```

Comment:
I would follow the convention set by `inferredTypesContainingFile` at line 578—give it a good variable name _and_ a good fake filename, because this shows up in `--traceResolution`.

Reply:
Agree, makes sense 👍🏻